script/ddgc_export_help.pl

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env perl
+
+use Moo;
+use MooX::Options;
+
+use File::Spec::Functions;
+use File::Path qw/ make_path /;
+use File::Copy;
+use IO::All -utf8;
+use HTML::TreeBuilder::LibXML;
+use Archive::Tar;
+use File::Find;
+use Try::Tiny;
+
+use FindBin;
+use lib $FindBin::Dir . "/../lib";
+
+has ddgc => ( is => 'lazy' );
+sub _build_ddgc {
+    require DDGC;
+    DDGC->new;
+}
+
+has now => (
+    is => 'ro',
+    default => sub { time() }
+);
+
+has ddgc_dir => ( is => 'lazy' );
+sub _build_ddgc_dir {
+    $_[0]->ddgc->config->rootdir_path
+}
+
+has help_dir => ( is => 'lazy' );
+sub _build_help_dir {
+    my ( $self ) = @_;
+    $self->_mkdir( $self->ddgc_dir, 'help' );
+}
+
+has pkg_dir => ( is => 'lazy' );
+sub _build_pkg_dir {
+    my ( $self ) = @_;
+    $self->_mkdir( $self->help_dir, $self->now );
+}
+
+has user => ( is => 'lazy' );
+sub _build_user {
+    my ( $self ) = @_;
+    $self->ddgc->find_user('ddgc');
+}
+
+has tarball => ( is => 'lazy' );
+sub _build_tarball {
+    my ( $self ) = @_;
+    catfile( $self->ddgc_dir, 'media', 'help-' . $self->now . '.tar.bz2' );
+}
+
+has archive_tar => ( is => 'lazy' );
+sub _build_archive_tar {
+    my ( $self ) = @_;
+    Archive::Tar->new;
+}
+
+sub _mkdir {
+    my ( $self, @dir ) = @_;
+    my $d = catdir( @dir );
+    make_path $d unless -d $d;
+    return $d;
+}
+
+sub dir_for {
+    my ( $self, $dir ) = @_;
+    my $d = catdir( $self->pkg_dir, $dir );
+    make_path $d unless -d $d;
+    return $d;
+}
+
+sub filename_for {
+    $_[1] =~ s{.*/}{}r;
+}
+
+sub write_markup {
+    my ( $self, $help ) = @_;
+    my $content = join " ",
+                  map { $_->content_html }
+                  $help->help_contents->all;
+    my $category = $help->help_category
+        ? $help->help_category->key
+        : 'hidden';
+    my $d = $self->dir_for( $category );
+
+    my $tree = HTML::TreeBuilder::LibXML->new;
+    $tree->parse( $content );
+    $tree->eof;
+
+    for my $node ( $tree->findnodes('//img') ) {
+        my $src = $node->attr('src');
+        if ( $src =~ m{^https?://}i ) {
+            my $m;
+            try {
+                $m = $self->ddgc->rs('Media')->create_via_url( $self->user, $src );
+            } catch {
+                warn "Downloading $src failed : $_";
+            };
+            next unless $m;
+            $src = '/media/' . $m->filename;
```

Comment:
Is there any chance `$m` could have a value but not have the filename property, eg if the download succeeds but it's garbled up or the content isn't what we expect? 
I guess it's a pretty slim chance and every other possible error is handled, so doesn't matter

Reply:
Should be OK - the [Media ResultSet's create call](https://github.com/duckduckgo/community-platform/blob/master/lib/DDGC/DB/ResultSet/Media.pm#L58) could probably use some guards 

Either way, the script is more likely to just crash than not have a filename - it's generated by all the md5 trash above that line and does not depend on the download's properties.